client/src/main/scala/com.gu.contentapi.client/Backoff.scala

```diff
@@ -0,0 +1,83 @@
+package com.gu.contentapi.client
+
+import java.util.concurrent.TimeUnit
+
+import com.gu.contentapi.client.model.{ ContentApiRecoverableException, HttpResponse }
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.Duration
+
+sealed abstract class Backoff extends Product with Serializable { self =>
+  val scheduledExecutor = new ScheduledExecutor(1)
```

Comment:
Ha, I did say to do that, but this is bad. Every time we create a `Backoff` (in `state`) a new thread pool will be created 😱 I suggest to make it an implicit parameter to `Backoff` and to move that line as an abstract member in `ContentApiClient`:
```scala
trait ContentApiClient {
  implicit val scheduler: ScheduledExecutor
}
```